src/components/Observations/Toolbar.js

```diff
@@ -1,53 +1,139 @@
 import IconMaterial from "react-native-vector-icons/MaterialIcons";
 import useCurrentUser from "sharedHooks/useCurrentUser";
+import useLocalObservations from "sharedHooks/useLocalObservations";
+import useNumUnuploadedObservations from "sharedHooks/useNumUnuploadedObservations";
+import useUploadObservations from "sharedHooks/useUploadObservations";
+import colors from "styles/tailwindColors";
 
 type Props = {
-  setView: Function
-  }
+  setView: Function,
+  view: string,
+};
 
-const Toolbar = ( { setView }: Props ): Node => {
+const Toolbar = ( { setView, view }: Props ): Node => {
   const currentUser = useCurrentUser( );
   const obsEditContext = useContext( ObsEditContext );
+  const { allObsToUpload } = useLocalObservations( );
+  const numUnuploadedObs = useNumUnuploadedObservations( );
+  const {
+    stopUpload,
+    uploadInProgress,
+    startUpload,
+    error: uploadError
+  } = useUploadObservations( allObsToUpload );
+
   const loading = obsEditContext?.loading;
   const syncObservations = obsEditContext?.syncObservations;
 
+  const totalObsToUpload = Math.max( allObsToUpload.length, numUnuploadedObs );
+  const progress = ( totalObsToUpload - numUnuploadedObs + 1 ) / ( totalObsToUpload + 1 );
+
+  const getSyncClick = ( ) => {
+    if ( numUnuploadedObs > 0 ) {
+      return startUpload;
+    }
+    return syncObservations;
+  };
+
+  const cleanProgress = ( progressToClean: number ) => {
+    if ( Number.isNaN( progressToClean ) || progressToClean < 0 ) {
+      return 0;
+    }
+
+    return progressToClean;
+  };
+
+  const getStatusText = ( ) => {
+    if ( !uploadInProgress && totalObsToUpload > 0 ) {
+      return t( "UPLOAD-X-OBSERVATIONS", { count: numUnuploadedObs } );
+    }
+    if ( totalObsToUpload > 0 ) {
+      return t( "Uploading-X-Observations", { count: numUnuploadedObs } );
+    }
+    return null;
+  };
+
+  const getSyncIconColor = ( ) => {
+    if ( uploadInProgress || totalObsToUpload > 0 ) {
+      return colors.inatGreen;
+    }
+    return colors.darkGray;
+  };
+
+  const statusText = getStatusText( );
+
+  /* eslint-disable react-native/no-inline-styles */
   return (
-    <View className="py-5 flex-row justify-between bg-white">
-      {currentUser ? (
-        <Pressable
-          onPress={syncObservations}
-          className="mx-3"
-          accessibilityRole="button"
-          disabled={loading}
-        >
-          <IconMaterial name="sync" size={30} />
-        </Pressable>
-      ) : (
-        <View className="mx-3" />
-      )}
-      {loading && <ActivityIndicator />}
-      <View className="flex-row mx-3">
-        <Pressable
-          onPress={( ) => setView( "list" )}
-          accessibilityRole="button"
-        >
-          <IconMaterial name="format-list-bulleted" size={30} />
-        </Pressable>
-        <Pressable
-          onPress={( ) => setView( "grid" )}
-          testID="ObsList.toggleGridView"
+    <View className="bg-white border-b border-[#e8e8e8]">
```

Comment:
Is there a way to style everything here without having 4 nested Views? I find this a bit difficult to read.